### PR TITLE
HPCC-26362 React MessageBox

### PR DESCRIPTION
**Pull request by GordonSmith**
Common up Modal code.

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jeclrsg Keep an eye out for incorrect MessageBox titles (lot of cut and paste) and Primary Button titles...

File: esp/src/src-react/components/MetricsOptions.tsx [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
 import nlsHPCC from "src/nlsHPCC";
 import { useMetricMeta, useMetricsOptions } from "../hooks/metrics";
-
-const dragOptions: IDragOptions = {
-    moveMenuItemText: "Move",
-    closeMenuItemText: "Close",
-    menu: ContextualMenu,
-};
+import { MessageBox } from "../layouts/MessageBox";
+import { pivotItemStyle } from "src-react/layouts/pivot";
```
Comment (jeclrsg) [inline esp/src/src-react/components/MetricsOptions.tsx:6]: "src-react" include path

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx [hunk 2]
```diff
@@ -123,31 +115,18 @@ export const BlobImportForm: React.FunctionComponent<BlobImportFormProps> = ({
...
-        </div>
-        <div className={componentStyles.body}>
-            <Stack>
-                <Controller
-                    control={control} name="destGroup"
-                    render={({
-                        field: { onChange, name: fieldName, value },
-                        fieldState: { error }
-                    }) => <TargetGroupTextField
+    return <MessageBox title={nlsHPCC.Import} show={showForm} setShow={closeForm}
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:118]: Wasn't sure if this was the kind of thing you meant about looking at titles. They previously included the type of the spray, but seems fine that they can all just read Import too.

**Review by jeclrsg — APPROVED** [review]

**Comment by ghalliday** [comment]
@GordonSmith will need rebasing before merging

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx (hunk 2 above)
Comment (GordonSmith) [reply esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx]: Yes it was!

**Comment by GordonSmith** [comment]
@ghalliday this has been rebased 
@jeclrsg I converted a few more "modals" over (new ones which were added) - can you re-review as needed?

**Review by jeclrsg — APPROVED** [review]

**Comment by GordonSmith** [comment]
@ghalliday ready for pull

**Comment by ghalliday** [comment]
@GordonSmith needs rebasing again - clashed with the label change.

**Comment by GordonSmith** [comment]
@ghalliday rebased

**Pull request merged**